frontend/test/metabase/scenarios/visualizations/pivot_tables.cy.spec.js

```diff
@@ -0,0 +1,164 @@
+import { signInAsAdmin, restore } from "__support__/cypress";
+import { SAMPLE_DATASET } from "__support__/cypress_sample_dataset";
+
+const { ORDERS, ORDERS_ID, PRODUCTS, PEOPLE } = SAMPLE_DATASET;
+
+const QUESTION_NAME = "Cypress Pivot Table";
+
+describe("scenarios > visualizations > pivot tables", () => {
+  beforeEach(() => {
+    restore();
+    signInAsAdmin();
+  });
+
+  it("should be created from an ad-hoc question", () => {
+    // We're intentionally setting the visualization to `bar` here
+    createAndVisitTestQuestion({ display: "bar" });
+
+    // By changing the visualization type, we make this an "ad-hoc" question
+    cy.findByText("Visualization").click();
+    cy.get(".Icon-pivot_table")
+      .should("be.visible")
+      .click();
+
+    cy.findByText(/Count by Users? → Source and Products? → Category/); // ad-hoc title
+
+    assertOnPivotSettings();
+    assertOnPivotTable();
+  });
+
+  it("should correctly display saved question", () => {
+    createAndVisitTestQuestion();
+
+    assertOnPivotTable();
+
+    // Open Pivot table side-bar
+    cy.findByText("Settings").click();
+
+    assertOnPivotSettings();
+  });
+
+  it("should not show sub-total data after a switch to other viz type", () => {
+    createAndVisitTestQuestion();
+
+    // Switch to "ordinary" table
+    cy.findByText("Visualization").click();
+    cy.get(".Icon-table")
+      .should("be.visible")
+      .click();
+
+    cy.contains(`Started from ${QUESTION_NAME}`);
+
+    cy.log("**-- Assertions on a table itself --**");
+    cy.get(".Visualization").within(() => {
+      cy.findByText(/Users? → Source/);
+      cy.findByText("783"); // Affiliate - Doohickey
+      cy.findByText("986"); // Twitter - Gizmo
+      cy.findByText(/Row totals/i).should("not.exist");
+      cy.findByText(/Grand totals/i).should("not.exist");
+      cy.findByText("3,520").should("not.exist");
+      cy.findByText("4,784").should("not.exist");
+      cy.findByText("18,760").should("not.exist");
+    });
+  });
+
+  it("should rearrange pivoted columns", () => {
+    createAndVisitTestQuestion();
+
+    // Open Pivot table side-bar
+    cy.findByText("Settings").click();
+
+    // Give it some time to open the side-bar fully before we start dragging
+    cy.findByText(/Pivot Table options/i);
+
+    // Drag the second aggregate (Product category) from table columns to table rows
```

Comment:
@paulrosenzweig 

Continuing from [this comment](https://github.com/metabase/metabase/pull/14051/files#r541291213) - are there other scenarios we should test?
- Drag row field into columns? (that gave me empty field values for the whole table)
- Assert that the `count` cannot be dragged anywhere?
- ... add your own scenario

Reply:
I don't think we need to. This checks that the setting interaction works, we requery for new data, and we update the chart. Also, the UI here might change a bit still so it'd better not too have too many tests that will break.

> that gave me empty field values for the whole table
Whoops! I either need to prevent that move or fix it's display.

Reply:
Ok, so this can be merged and please let me know if you think of any more test cases, or need help reproing anything related.

Thank you for reviewing this.